corehq/util/pagination.py

```diff
@@ -265,9 +265,6 @@ def __iter__(self):
         except StopToResume:
             return
 
-        self.state.args = None
-        self.state.kwargs = None
-        self.state.complete = True
```

Comment:
Yes, in this sense a resumable iteration is now never really "complete" since it can always be resumed, and more items will be yielded if they exist. For a given iteration run there is still a meaningful concept of "complete" that happens when all existing items have been yielded; at that point it raises `StopIteration` like any other iterator.

The change here is that previously we were discarding iteration state when the iteration was exhausted the first time. We are no longer discarding that information. Going forward, the concepts of completion and whether it is meaningful to resume after completion depends on iteration context: is it possible for new items to be added at the "end," and does it make sense to process those new items after the initial iteration has completed?

Examples of iterations with different answers to those questions:
- Forms & cases migration: if new couch forms are submitted after the iteration has "completed" then it makes sense to resume and process them. It is usually not possible to submit new forms in Couch while the "end" of the iteration is being processed, but in the case where the migration cannot be committed and switched over to SQL it is useful to re-enable form submissions in Couch and resume the migration/iteration later. This is possible since the forms iteration is ordered by date.
- Migrate Couch attachments to BlobDB: "attachments" for new documents are created in the blobdb rather than in Couch; this code is enabled before the migration is started. Once the migration has iterated through all documents (presumably ordered by doc_id) and processed all Couch attachments it would not make sense to resume iteration since any new documents encountered at the "end" would have all attachments in the blobdb.